File: src/things/RockLichen.cpp
#include "things/RockLichen.h"

#include "App.h"

RockLichen::RockLichen(UUID id) : Blob("rock lichen", id)
{
  //ctor
}

RockLichen::~RockLichen()
{
  //dtor
}

std::string RockLichen::_get_description() const
{
  return "rock lichen";
}

sf::Vector2u RockLichen::get_tile_sheet_coords(int frame) const
{
  return sf::Vector2u(4, 5);
}

File: resources/things/RockLichen.lua
-- Definition of special functions for the RockLichen object type.

require "resources/things/Blob"

RockLichen = inheritsFrom(Blob, "RockLichen")
RockLichen.intrinsics.name = "rock lichen"
RockLichen.intrinsics.plural = "rock lichens"
RockLichen.intrinsics.creatable = true

RockLichen.intrinsics.maxhp = range(1, 1)

function RockLichen.get_description()
	return "A lichen that normally grows on a rock."
end
